lib/trino-orc/src/main/java/io/trino/orc/metadata/statistics/StringStatisticsBuilder.java

```diff
@@ -158,9 +173,19 @@ public ColumnStatistics buildColumnStatistics()
         return stringStatisticsBuilder.buildStringStatistics();
     }
 
-    private Slice dropStringMinMaxIfNecessary(Slice minOrMax)
+    private Slice dropStringMinMaxIfNecessary(Slice minOrMax, boolean isMin)
     {
         if (minOrMax == null || minOrMax.length() > stringStatisticsLimitInBytes) {
+            if (minOrMax != null && shouldCompactMinMax) {
+                if (isMin) {
```

Comment:
This can be written with less nesting

```java
        if (minOrMax == null || (!shouldCompactMinMax && minOrMax.length() > stringStatisticsLimitInBytes)) {
            return null;
        }
        if (minOrMax.length() > stringStatisticsLimitInBytes) {
            if (isMin) {
                byte[] min = StringCompactor.computeMin(minOrMax.getBytes(), stringStatisticsLimitInBytes);
                return Slices.wrappedBuffer(min);
            }
            byte[] max = StringCompactor.computeMax(minOrMax.getBytes(), stringStatisticsLimitInBytes);
            return Slices.wrappedBuffer(max);
        }
```